File: Behavioral/Observer/observer.py
class Subject(ABC):
    @abstractmethod
    def attach(self, observer): pass
    @abstractmethod
    def notify(self): pass

class ConcreteSubject(Subject):
    _observers = []
    def attach(self, observer): self._observers.append(observer)
    def notify(self):
        for observer in self._observers:
            observer.update(self)

File: Behavioral/Observer/observer.cs
public interface IObserver { void Update(ISubject subject); }

public interface ISubject {
    void Attach(IObserver observer);
    void Notify();
}

class Subject : ISubject {
    private List<IObserver> _observers = new List<IObserver>();
    public void Attach(IObserver observer) => _observers.Add(observer);
    public void Notify() {
        foreach (var observer in _observers) observer.Update(this);
    }
}